### PR TITLE
Support for showing local variables for interpreted functions in slime

### PR DESCRIPTION
**Pull request by alanruttenberg**
A version without all the commit noise.

### CODE REVIEW

File: contrib/abcl-introspect/abcl-introspect.lisp [hunk 1]
```diff
@@ -364,7 +364,180 @@ above have used annotate local functions"
                                              &allow-other-keys)
   (annotate-clos-slots (mop::class-direct-slots (find-class name))))
 
+;; Environments
+
+;; Return a list of the variables and functions in an environment. The form of the list is
+;; (kind name value)
+;; where kind is either :lexical-variable or :lexical-function :special-variable
+
+(defun environment-parts(env)
```
Comment (easye) [inline contrib/abcl-introspect/abcl-introspect.lisp:373]: Whitespace

**Comment by easye** [comment]
How do I test this in SLIME?

I evaluated the subsequent form, expecting to be able to inspect `x` and `y` when the `break` brought me into SLIME, but I don't see anything for locals other than `a` and `b`.  Is there perhaps an associated patch to SLIME?
```
(defun add (a b)                                                                                                              
  (let ((x a)                                                                                                                 
        (y b))                                                                                                                
    (break)                                                                                                                   
    (+ x y)))  
```

**Comment by alanruttenberg** [comment]
Yes. I put in a pull request against waste/slime. https://github.com/easye/slime/pull/2

**Comment by easye** [comment]
This isn't the branch where I cleanup up the whitespace, changed the exported symbol prefix, and added the test.  Please close this branch, and use that one for this line of development.

**Comment by alanruttenberg** [comment]
And which one would that be? Master, I presume?

**Comment by easye** [comment]
<https://github.com/easye/abcl/tree/alanr/inspectable-locals/20220217b>

**Comment by easye** [comment]
Superseded by <https://github.com/armedbear/abcl/pull/449>

**Pull request closed without merging**